src/pages/EditMonitor.vue

```diff
@@ -486,6 +495,16 @@ export default {
         async submit() {
             this.processing = true;
 
+            // Check if friendly name has been supplied. If not, use URL
+            // or hostname
+            if (this.monitor.name === "") {
+                if (this.monitor.hostname === null || this.monitor.hostname === "") {
+                    this.monitor.name = this.monitor.url;
+                } else {
+                    this.monitor.name = this.monitor.hostname;
+                }
```

Comment:
You can use `this.monitor.name = this.monitor.hostname || this.monitor.url;`

Reply:
Fixed